src/Uno.UWP/Globalization/NumberFormatting/StringBuilderPool.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Text;
+
+namespace Uno.Globalization.NumberFormatting;
+
+internal class StringBuilderPool
+{
+	private StringBuilder[] _items;
+
+	public static StringBuilderPool Instance { get; } = new StringBuilderPool();
+
+	public StringBuilderPool()
+	{
+		_items = new StringBuilder[Environment.ProcessorCount * 2];
+	}
+
+	public StringBuilder Get()
+	{
+		for (int i = 0; i < _items.Length; i++)
+		{
+			var val = _items[i];
+			if (val != null)
+			{
+				_items[i] = null;
+				return val;
+			}
+		}
+
+		return new StringBuilder();
+	}
+
+	public void Return(StringBuilder obj)
+	{
+		obj.Clear();
+
+		for (int i = 0; i < _items.Length; i++)
+		{
+			if (_items[i] == null)
+			{
+				_items[i] = obj;
+				return;
+			}
+		}
+	}
```

Comment:
The `StringBuilderPool` is not thread-safe. We can copy & paste from[ Microsoft.Extensions.ObjectPool](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.objectpool) is it good idea?

Reply:
There's no need at this point. If we find that this is a performance concern, we'll adjust it at this time.

You can make a `StringBuilder` field marked with `[ThreadStatic]` and make sure to clear the instance it on each formatting call, it'll be good enough for our scenario.

Reply:
As the [Documentation](https://docs.microsoft.com/en-us/dotnet/api/system.threadstaticattribute?view=net-6.0#remarks) says:

> Do not specify initial values for fields marked with ThreadStaticAttribute, because such initialization occurs only once, when the class constructor executes, and therefore affects only one thread. If you do not specify an initial value, you can rely on the field being initialized to its default value if it is a value type, or to null if it is a reference type.

if we use `ThreadStaticAttribute` on other threads the value of `StringBuilder` would be null. `[ThreadLocal]`(https://docs.microsoft.com/en-us/dotnet/api/system.threading.threadlocal-1?view=net-6.0) solves this issue. Do you agree to use `ThreadLocal`?

Reply:
Either one is good. `ThreadLocal` works as well.

Reply:
Maybe you could move StringBuilderPool to Uno's BCL so you can use it elsewhere.

Reply:
Good idea but I think it would be dangerous because currently there isn't any way to ensure that StringBuilder1 is used by another code so being in `Windows.Globalization.NumberFormatting` is now better. To avoid the issue we should use ObjectPool as you said first.